crates/storage/engines/api.rs

```diff
@@ -165,4 +172,11 @@ pub trait StoreEngine: Debug + Send {
     // Obtain a world state from an empty root
     // This method should be used when creating the genesis world state
     fn new_state_trie(&self) -> Result<Trie, StoreError>;
+
+    // Get the canonical block hash for a given block number.
+    fn set_canonical_block_hash(
```

Comment:
nit: `set_canonical_block` ? You're setting the canonical block here, the hash is just the representation of the block